manifests/config.pp

```diff
@@ -53,7 +59,17 @@
       $changes = "clear ${name}"
     }
     default: {
-      fail "Unknown value for ensure '${ensure}'"
+      case $value {
+        /^.+$/: {
+          $changes = "set ${name} '${value}'"
+        }
+        /^$/: {
+          $changes = "clear ${name}"
+        }
+        default: {
+          $changes = "rm ${name}"
+        }
+      }
```

Comment:
This whole block is now equivalent to:

```
if $value == "" {
  $changes = "clear ${name}"
} else {
  $changes = "set ${name} '${value}'"
}
```

Reply:
Old selector was intentionally left to have backward compatibility. If you want to use old style you use `ensure = Enum[present, blank, absent]`. However if you want to create a flexible and short profile, you can omit ensure to allow for value analyze. I did not want to change initial logic completely, since some people would still like to use that.

`ensure => present, value => 'foo'` _set ${name} '${value}'_
`ensure => present, value => ''` _error as before_
`ensure => present, value => undef` _error as before_
`ensure => blank` _clear ${name}_
`ensure => absent` _rm ${name}_

`value => 'foo'` _set ${name} '${value}'_
`value => ''` _clear ${name}_
`value => undef` _rm ${name}_

Reply:
OK. I understand the need and the implementation, but I'm trying to think of a better way... I have a few problems with this PR:

* it duplicates the interface for one action (two ways of doing the same things, and much more complexity to maintain that seemingly simple code)
* it changes the default (currently `ensure => present` ensures to `set`) to ensuring absent values (since `ensure => undef` now defaults to `rm`)
* I can't think of a single resource type that takes `ensure => undef`. It just doesn't make much sense. It looks like your proposal is equivalent to defaulting to `ensure => absent` actually, which is a breaking change.

Reply:
Hm. How about to add `ensure => variant` for example and only with this `ensure` act according to `value` type/size ?